src/main/java/com/synopsys/integration/alert/database/api/configuration/DescriptorAccessor.java

```diff
@@ -374,62 +376,45 @@ private RegisteredDescriptorEntity findDescriptorByName(final String name) throw
-                       .findById(id)
-                       .orElseThrow(() -> new AlertDatabaseConstraintException("A descriptor with that id did not exist"));
+                   .findById(id)
+                   .orElseThrow(() -> new AlertDatabaseConstraintException("A descriptor with that id did not exist"));
     }
 
-    // TODO implement a join
-    private List<DefinedFieldModel> getFieldsForDescriptorId(final Long descriptorId, final ConfigContextEnum context) throws AlertDatabaseConstraintException {
-        final Set<Long> fieldIds = descriptorFieldRepository
-                                           .findByDescriptorId(descriptorId)
-                                           .stream()
-                                           .map(DescriptorFieldRelation::getFieldId)
-                                           .collect(Collectors.toSet());
-        if (fieldIds.isEmpty()) {
-            return Collections.emptyList();
-        }
-
-        final Long contextId = saveContextAndReturnId(context);
-        fieldContextRepository
-                .findByContextId(contextId)
-                .stream()
-                .filter(relation -> fieldIds.contains(relation.getFieldId()))
-                .map(FieldContextRelation::getFieldId)
-                .collect(Collectors.toList());
+    private List<DefinedFieldModel> getFieldsForDescriptorId(final Long descriptorId, final Long contextId, ConfigContextEnum context) throws AlertDatabaseConstraintException {
+        List<DefinedFieldModel> fields = definedFieldRepository.findByDescriptorIdAndContext(descriptorId, contextId)
+                                             .stream()
+                                             .map(entity -> new DefinedFieldModel(entity.getKey(), context, entity.getSensitive()))
+                                             .collect(Collectors.toList());
 
-        return definedFieldRepository
-                       .findAllById(fieldIds)
-                       .stream()
-                       .map(entity -> new DefinedFieldModel(entity.getKey(), context, entity.getSensitive()))
-                       .collect(Collectors.toList());
+        return fields;
     }
 
     private Set<ConfigContextEnum> getContextsForFieldId(final Long fieldId) {
         return fieldContextRepository
-                       .findByFieldId(fieldId)
-                       .stream()
-                       .map(FieldContextRelation::getContextId)
-                       .flatMap(configId -> configContextRepository.findAllById(Collections.singleton(configId)).stream())
-                       .map(entity -> ConfigContextEnum.valueOf(entity.getContext()))
-                       .collect(Collectors.toSet());
+                   .findByFieldId(fieldId)
+                   .stream()
+                   .map(FieldContextRelation::getContextId)
+                   .flatMap(configId -> configContextRepository.findAllById(Collections.singleton(configId)).stream())
```

Comment:
Collections.singleton(...) can now be replaced by Set.of(...) while you're in there ;)